LiveSched/src/main/java/dev/coms4156/project/livesched/Task.java

```diff
@@ -0,0 +1,146 @@
+package dev.coms4156.project.livesched;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+/**
+ * Represents a task that has to be done. 
+ * This class stores the id, resources needed, and the location of the task.
+ */
+public class Task {
+  private final String taskId;
+  private Map<ResourceType, Integer> resourceList; // Key = ResoureceType, Value = Units needed
+  private int priority; // value between 1 and 5
+  private LocalDateTime startTime;
+  private LocalDateTime endTime;
+  private double latitude;
+  private double longitude;
+
+  /**
+   * Constructs a new Task object with the given parameters.
+   *
+   * @param taskId        the unique ID of the resource
+   * @param resourceList  the map of resources and their quantity needed
+   * @param priority      the priority of the task
+   * @param startTime     the time that task starts
+   * @param endTime       the time that task ends
+   * @param latitude      the latitude of the task's location
+   * @param longitude     the longitude of the task's location
+   * @throws IllegalArgumentException if the {@code resourceId} is null or empty,
+   *                                  or if the latitude or longitude is out of bounds
+   */
+  public Task(String taskId, Map<ResourceType, Integer> resourceList, int priority, 
+        LocalDateTime startTime, LocalDateTime endTime, double latitude, double longitude) {
+    if (taskId == null || taskId.trim().isEmpty()) {
+      throw new IllegalArgumentException("Task ID cannot be null or empty.");
+    }
+    validateLatLong(latitude, longitude);
```

Comment:
Since we're using this for both Resources and Task classes, I feel like there should be a separate Location class that stores this information (latitude, longitude) and have a reusable method for `validateLatLong`. But for now, this is our initial implementation so maybe we can change as things progress. It looks good overall.

Reply:
I agree!

Reply:
Added a TODO comment for future reference :)

Reply:
Thanks!